dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -120,6 +120,13 @@ def join
       return
     end
     result = @section.add_student current_user
+    # add_student returns 'failure' when id of current user is owner of @section
+    if result == 'failure'
```

Comment:
Worth noting that `add_student` returns `'failure'` in the Section model [here](https://github.com/code-dot-org/code-dot-org/blob/e952410338380e2c0a0b74dad50f70df9b9b4813/dashboard/app/models/sections/section.rb#L164).